### PR TITLE
Bump env

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

Note: I'm looking into why an lcm update was required. I think it's due to ordering and we should stop running into this once https://github.com/stellar/stellar-core/pull/4029/ and the meta normalization change is merged.

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

**Review by SirTyson — APPROVED** [review]
This behavior is expected. We've recompiled the test Wasms since the last env bump and the size of the binaries has changed. It looks like the TXs are all the same, but the refundable fees have changed slightly due to the size difference. Of course, the order of all the TXs changed too because we don't normalize. I haven't done a rigorous diff analysis. just manually comparing the TXs, so take this with a grain of salt, but I think everything looks fine.

File: src/rust/src/contract.rs [hunk 1]
```diff
@@ -157,9 +157,9 @@ impl From<xdr::Error> for CoreHostError {
 impl std::error::Error for CoreHostError {}
 
 fn non_metered_xdr_from_cxx_buf<T: ReadXdr>(buf: &CxxBuf) -> Result<T, HostError> {
-    Ok(T::read_xdr(&mut xdr::DepthLimitedRead::new(
+    Ok(T::read_xdr(&mut xdr::Limited::new(
         Cursor::new(buf.data.as_slice()),
-        MARSHALLING_STACK_LIMIT,
+        Limits {depth: MARSHALLING_STACK_LIMIT, len: 0x1000000},
```
Comment (SirTyson) [inline src/rust/src/contract.rs:162]: Use the buffer size here for the length.

Comment (sisuresh) [reply src/rust/src/contract.rs]: fixed

File: src/rust/src/contract.rs [hunk 2]
```diff
@@ -168,9 +168,9 @@ fn non_metered_xdr_from_cxx_buf<T: ReadXdr>(buf: &CxxBuf) -> Result<T, HostError
 
 fn non_metered_xdr_to_rust_buf<T: WriteXdr>(t: &T) -> Result<RustBuf, HostError> {
     let mut vec: Vec<u8> = Vec::new();
-    t.write_xdr(&mut xdr::DepthLimitedWrite::new(
+    t.write_xdr(&mut xdr::Limited::new(
         Cursor::new(&mut vec),
-        MARSHALLING_STACK_LIMIT,
+        Limits {depth: MARSHALLING_STACK_LIMIT, len: 0x1000000},
```
Comment (dmkozh) [inline src/rust/src/contract.rs:173]: This is hard to interpret and also probably too much. I would probably set this to 5 MB and write that clearly as `5 * 1024 * 1024`

Comment (sisuresh) [reply src/rust/src/contract.rs:173]: Is 5MB enough as an upper bound?

Comment (sisuresh) [reply src/rust/src/contract.rs:173]: I'm thinking specifically about WASM size

Comment (dmkozh) [reply src/rust/src/contract.rs:173]: The LE size limit is 64K, so we're not even close to that limit.
More importantly, this is non-metered function. It is only used in two contexts now: for diagnostics (non-observable, can be updated without protocol change if ever necessary), and for TTL entry encoding (small bounded size, though even that seems a bit iffy). Given that it's non-metered, I don't think we should use it to convert any non-constant-size observable data.

Comment (sisuresh) [reply src/rust/src/contract.rs]: Updated

**Comment by dmkozh** [comment]
r+ d958347

**Comment by sisuresh** [comment]
@dmkozh I had to fix a merge conflict.

**Comment by dmkozh** [comment]
r+ 6046c8e

**Pull request merged**